tmail-backend/webadmin/webadmin-rate-limit/src/test/java/com/linagora/tmail/webadmin/RateLimitPlanUsersRoutesTest.java

```diff
@@ -0,0 +1,166 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package com.linagora.tmail.webadmin;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.jetty.http.HttpStatus.BAD_REQUEST_400;
+import static org.eclipse.jetty.http.HttpStatus.NOT_FOUND_404;
+import static org.eclipse.jetty.http.HttpStatus.OK_200;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Map;
+import java.util.UUID;
+
+import org.apache.james.core.Username;
+import org.apache.james.user.api.UsersRepository;
+import org.apache.james.user.api.UsersRepositoryException;
+import org.apache.james.webadmin.WebAdminServer;
+import org.apache.james.webadmin.WebAdminUtils;
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import com.linagora.tmail.rate.limiter.api.InMemoryRateLimitationPlanRepository;
+import com.linagora.tmail.rate.limiter.api.RateLimitationPlanRepository;
+import com.linagora.tmail.rate.limiter.api.RateLimitingPlanId;
+import com.linagora.tmail.rate.limiter.api.RateLimitingPlanUserRepository;
+import com.linagora.tmail.rate.limiter.api.memory.MemoryRateLimitingPlanUserRepository;
+
+import io.restassured.RestAssured;
+import io.restassured.path.json.JsonPath;
+import reactor.core.publisher.Mono;
+
+public class RateLimitPlanUsersRoutesTest {
+    private static final String BASE_PATH = "/rate-limit/plans";
+    private static final String ADD_USER_TO_PLAN_PATH = "/rate-limit/plans/%s/users/%s";
```

Comment:
Users is the base resource here that rate limiting plans are attached to.

THus it should come first in the URL.

`/users/bob@domain.tld/rate-limiting-plan` to get the plan of a user for instance.

`/rate-limit-plans/%s/users` is OK to list the users attached to a plan (as plan is here the base resource)